### ISSUE TITLE
Why use the `.api` extension and not `.abi`?

### ISSUE THREAD

**Issue by joffrey-bion** [Kotlin/binary-compatibility-validator#198]
Given that these files are meant to represent a serialized ABI to check binary compatibility, and that the tool is called `binary-compatibility-validator` it would make sense to call this the ABI, right?

**Comment by qwwdfsad** [comment]
It indeed makes sense. This is rather a historical oddity where we first started with "what is public API" and then every other name just followed.
If we ever consider pushing BCV to 1.0 or merging it into Kotlin upstream, we definitely should consider the renaming

**Comment by joffrey-bion** [comment]
I see, thanks. I was thinking that maybe because of the change in structure with the klib / KMP support, it might be a good time to change the extension too

**Comment by fzhinkin** [comment]
We thought about that too, but changing the name of JVM dump files would be painful for existing users, and having two files (for KLib and class files) with the (conceptually) same stuff inside but different names looked weird.

**Comment by joffrey-bion** [comment]
> changing the name of JVM dump files would be painful for existing users

In which way?

These files are updated each time an ABI change happens, so it's already part of the process to update them/commit them in some way. Making a source code change + `apiDump` + commit is as complicated as making a plugin version update + `apiDump` + commit. Git is quite good at recognizing the file even with a different name if the contents don't change. Is there something I am missing?

**Comment by fzhinkin** [comment]
> In which way?

- If we change the file extension name, plugin's update in a project will make the dump invalid (well, the dump will not exists from plugin's POV) even if nothing had changed in the ABI;
- `git --follow`  helps with renaming, but it's not something working out of the box.

Given that the new release introduces experimental feature useful for some developers (those who owns KMP projects), it's hard to justify why everyone else should be disturbed.

As Vsevolod said, it's something that could be done with a major release, but for now we decided to stick to existing naming convention.

**Comment by joffrey-bion** [comment]
> plugin's update in a project will make the dump invalid

Yes, and the good thing about it is that it's a conscious decision, so nothing surprising to the users here. It has happened several times in the past that updating Kotlin or the plugin breaks the dumps, and at least for me personally it has never been more painful than forgetting to `apiDump` after a code change.

**Comment by shanshin** [comment]
As part of [the migration of separate BCV functionality to Kotlin Gradle Plugin](https://youtrack.jetbrains.com/issue/KT-71098/Migrate-Binary-Compatibility-Validator-to-Kotlin-Gradle-Plugin), the addition of new features to a separate plugin has been discontinued.

It will be developed a format that eliminates this problem there: https://youtrack.jetbrains.com/issue/KT-78009
